src/hotspot/share/runtime/objectMonitor.hpp

```diff
@@ -285,50 +280,60 @@ class ObjectMonitor : public CHeapObj<mtObjectMonitor> {
 
   // Returns true if this OM has an owner, false otherwise.
   bool      has_owner() const;
-  int64_t   owner() const;  // Returns null if DEFLATER_MARKER is observed.
+  int64_t   owner() const;  // Returns NO_OWNER if DEFLATER_MARKER is observed.
   int64_t   owner_raw() const;
 
-  static int64_t owner_for(JavaThread* thread);
-  static int64_t owner_for_oop(oop vthread);
+  // These methods return the value we set in _owner when acquiring
+  // the monitor with the given thread/vthread (tid).
+  static int64_t owner_from(JavaThread* thread);
+  static int64_t owner_from(oop vthread);
 
   // Returns true if owner field == DEFLATER_MARKER and false otherwise.
   bool      owner_is_DEFLATER_MARKER() const;
   // Returns true if 'this' is being async deflated and false otherwise.
   bool      is_being_async_deflated();
-  // Clear _owner field; current value must match old_value.
-  void      release_clear_owner(JavaThread* old_value);
+  // Clear _owner field; current value must match thread's tid.
+  void      release_clear_owner(JavaThread* thread);
   // Simply set _owner field to new_value; current value must match old_value.
   void      set_owner_from_raw(int64_t old_value, int64_t new_value);
+  // Same as above but uses tid of current as new value.
   void      set_owner_from(int64_t old_value, JavaThread* current);
-  // Simply set _owner field to current; current value must match basic_lock_p.
+  // Set _owner field to tid of current thread; current value must be ANONYMOUS_OWNER.
   void      set_owner_from_BasicLock(JavaThread* current);
```

Comment:
Shouldn't tid there be the basicLock?

Reply:
Right I understand we can't store the BasicLock* directly in owner, but the naming of this method has me confused as to what it actually does. With the old version we have:

Before: owner = BasicLock* belonging to current
After: owner = JavaThread* of current

with the new version we have:

Before: owner = ANONYMOUS_OWNER
After: owner = tid of current

so "BasicLock" doesn't mean anything here any more. Isn't this just `set_owner_from_anonymous` ?

Reply:
I see your point. I removed this method and had the only caller just call set_owner_from_anonymous() and set_stack_locker(nullptr). There was one other caller in ObjectMonitor::complete_exit() but it was actually not needed so I removed it. ObjectMonitor::complete_exit() is only called today on JavaThread exit to possibly unlock monitors acquired through JNI that where not unlocked.